### PR TITLE
Refactor/merge ofrak components

### PR DESCRIPTION
**Pull request by EdwardLarson**
**Link to Related Issue(s)**
One of the main reasons `ofrak_components` was split out originally was to separate those dependency-heavy components from the core OFRAK. With the dependency handling recently merged in, we can streamline OFRAK by combining `ofrak_components` into `core_ofrak` without making it harder to install OFRAK.

**Please describe the changes in your request.**

The `ofrak_components` package is removed and its contents moved into `ofrak.core`

**Anyone you think should look at this, specifically?**

@whyitfor

### CODE REVIEW

**Comment by rbs-jacob** [comment]
One very small change I've been meaning to make that could be in scope for this MR is adding the built `ofrak_core/ofrak/core/entropy/entropy.so.1` file to `.gitignore` at the root of the repo. That way the generated file in the repo doesn't accidentally get committed (or deleted with `git clean -f -d`). 

Would you be able to squeeze that in here @EdwardLarson?

**Comment by EdwardLarson** [comment]
> adding the built `ofrak_core/ofrak/core/entropy/entropy.so.1` file to `.gitignore` at the root of the repo

@rbs-jacob done!

File: docs/user-guide/component/packer.md [hunk 1]
```diff
@@ -53,9 +54,10 @@ await resource.run(ZipPacker)
 
 ### Run Automatically
 Packers can also be run automatically against resources with valid resource tags. For example, consider the following code:
+
 ```python
 from ofrak.resource import Resource
-from ofrak_components.zip import ZipArchive
+from ofrak.core.zip import ZipArchive
```
Comment (rbs-jacob) [inline docs/user-guide/component/packer.md:60]: These imports also almost certainly have to be fixed in the GUI. Lines like the following will need to be changed:

https://github.com/redballoonsecurity/ofrak/blob/e1386b54a08a133d63f03726248cfc76f3f19744/frontend/src/AttributesView.svelte#L26

**Review by rbs-jacob — CHANGES_REQUESTED** [review]
Looks mostly good! Two minor suggestions, plus there are still a bunch of places (mostly in the docs) that seem to refer to `ofrak_components`.

```
jacob@jacob ofrak % rg ofrak_component
disassemblers/ofrak_binary_ninja/LICENSE
30:ofrak_components, ofrak_io, ofrak_type, ofrak_patch_maker, ofrak_angr,

disassemblers/ofrak_ghidra/LICENSE
30:ofrak_components, ofrak_io, ofrak_type, ofrak_patch_maker, ofrak_angr,

disassemblers/ofrak_angr/LICENSE
30:ofrak_components, ofrak_io, ofrak_type, ofrak_patch_maker, ofrak_angr,

ofrak_io/LICENSE
30:ofrak_components, ofrak_io, ofrak_type, ofrak_patch_maker, ofrak_angr,

ofrak_tutorial/LICENSE
30:ofrak_components, ofrak_io, ofrak_type, ofrak_patch_maker, ofrak_angr,

disassemblers/ofrak_capstone/LICENSE
30:ofrak_components, ofrak_io, ofrak_type, ofrak_patch_maker, ofrak_angr,

LICENSE
30:ofrak_components, ofrak_io, ofrak_type, ofrak_patch_maker, ofrak_angr,

docs/gen_ref_nav.py
70:generate_docs(["ofrak_components"], "ofrak_components")

docs/environment-setup.md
220:    for d in ofrak_io ofrak_type ofrak_patch_maker ofrak_components ofrak; do make -C "${d}" develop; done

ofrak_type/LICENSE
30:ofrak_components, ofrak_io, ofrak_type, ofrak_patch_maker, ofrak_angr,

ofrak_patch_maker/LICENSE
30:ofrak_components, ofrak_io, ofrak_type, ofrak_patch_maker, ofrak_angr,

frontend/src/ByteclassView.svelte
41:      resource.get_attributes()["ofrak_components.entropy.entropy.DataSummary"];
52:        "ofrak_components.entropy.entropy.DataSummary"

docs/user-guide/flash_component.md
43:This format is interesting because it has a different sized tail block as well as different delimiters to represent the type of block. The [FlashUnpacker][ofrak_components.flash.FlashResourceUnpacker] is able to handle these different types of fields by providing attributes in [FlashAttributes][ofrak_components.flash.FlashAttributes] that also includes a [FlashEccAttributes][ofrak_components.flash.FlashEccAttributes]. We will describe the other parts of these attributes and how to use them later in this page.
46:The class [FlashFieldType][ofrak_components.flash.FlashFieldType] contains field types that are commonly encountered in flash dumps:
74:A [FlashAttributes][ofrak_components.flash.FlashAttributes] must be provided in order to use the flash component. As with other aspects of OFRAK, this can be modified and overridden if it does not work specifically for your use case.
77:The FlashAttributes][ofrak_components.flash.FlashAttributes] is necessary for communicating the information necessary to understand your specific flash format.
91:The `ecc_attributes` are also important for any dumps that include ECC. You have the option of providing the algorithms for encoding, decoding, and correcting the data. In addition, this is where the magic and any delimiter bytes are specified. See [FlashEccAttributes][ofrak_components.flash.FlashEccAttributes] for more information.
95:The Flash components can be used like any other OFRAK components. The first step is to tag a resource as a [FlashResource][ofrak_components.flash.FlashResource] and tag it with its flash resource attributes:

frontend/src/AttributesView.svelte
26:        "ofrak_components.entropy.entropy.DataSummary",

docs/contributor-guide/component/external_tools.md
15:components which do not use an external tool). The [ZipUnpacker][ofrak_components.zip.ZipUnpacker] is an example:
84:cover certain edge cases. For example, [SquashfsUnpacker][ofrak_components.squashfs.SquashfsUnpacker] requires

docs/user-guide/component/unpacker.md
44:from ofrak_components.elf.unpacker import ElfUnpacker

ofrak_core/MANIFEST.in
2:recursive-include ofrak_components/entropy/ *

frontend/src/MagnitudeView.svelte
41:      resource.get_attributes()["ofrak_components.entropy.entropy.DataSummary"];
52:        "ofrak_components.entropy.entropy.DataSummary"

ofrak_core/ofrak/ofrak_cli.py
18:        self._ofrak_components: Optional[Dict[str, Type[ComponentInterface]]] = None
22:        self._ofrak_component_dependencies: Optional[
47:        if self._ofrak_components is None:
48:            self._ofrak_components = {}
51:                    self._ofrak_components[component.__name__] = component
53:        return self._ofrak_components
84:        if self._ofrak_component_dependencies is None:
85:            ofrak_components = self.components.values()
86:            self._ofrak_component_dependencies = {
87:                c: list(c.external_dependencies) for c in ofrak_components  # type: ignore
89:        return self._ofrak_component_dependencies

docs/user-guide/component/analyzer.md
43:from ofrak_components.magic import Magic
57:from ofrak_components.magic import MagicAnalyzer

ofrak_core/LICENSE
30:ofrak_components, ofrak_io, ofrak_type, ofrak_patch_maker, ofrak_angr,

ofrak_core/test_ofrak/unit/test_resource.py
187:def mock_ofrak_component(ofrak):

frontend/src/EntropyView.svelte
41:      resource.get_attributes()["ofrak_components.entropy.entropy.DataSummary"];
52:        "ofrak_components.entropy.entropy.DataSummary"
```

**Comment by EdwardLarson** [comment]
@rbs-jacob I removed the hanging references in the docs and frontend code, good catch! I left the references in the LICENSE though - best not to touch those as far as I'm concerned, and shouldn't do any harm to mention that package.

**Review by rbs-jacob — APPROVED** [review]
Looks good to me! Merge whenever you're ready.

**Pull request merged**